diff --git a/Makefile-gir.am b/Makefile-gir.am
index 919e2697..857c5082 100644
--- a/Makefile-gir.am
+++ b/Makefile-gir.am
@@ -213,7 +213,8 @@ GIRepository_2_0_gir_FILES = \
         $(top_srcdir)/girepository/gitypelib.h	    \
         $(top_srcdir)/girepository/gitypes.h
 
-BUILT_GIRSOURCES += GIRepository-2.0.gir
+# GIRepository fails to find it's own library. Skip for now as we do not use this .gir
+# BUILT_GIRSOURCES += GIRepository-2.0.gir
 
 CLEANFILES += $(BUILT_GIRSOURCES)
 
