diff --git a/configure.ac b/configure.ac
index 002681a..9bb855e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -232,16 +232,7 @@ INTROSPECTION_COMPILER=`$PKG_CONFIG --variable=g_ir_compiler gobject-introspecti
 AC_SUBST(INTROSPECTION_SCANNER)
 AC_SUBST(INTROSPECTION_COMPILER)
 
-# compiler warnings, errors, required cflags, and code coverage support
-GNOME_COMPILE_WARNINGS([maximum])
-AC_MSG_CHECKING(for Gnome code coverage support)
-m4_ifdef([GNOME_CODE_COVERAGE],
-         [AC_MSG_RESULT(yes)
-          GNOME_CODE_COVERAGE],
-         [AC_MSG_RESULT(no)
-          GNOME_CODE_COVERAGE_RULES=''
-          AC_SUBST([GNOME_CODE_COVERAGE_RULES])
-          enable_code_coverage="no"])
+enable_code_coverage="no"
 
 if test "x$GCC" = "xyes"; then
   JH_ADD_CFLAG([-Wall])
--- a/Makefile.am
+++ b/Makefile.am
@@ -148,4 +148,3 @@ check.nemiver:
 check.valgrind:
        cd tests && $(MAKE) check.valgrind
 
-@GNOME_CODE_COVERAGE_RULES@
